cxplat/src/memory.c

```diff
@@ -2,39 +2,28 @@
 // SPDX-License-Identifier: MIT
 
 #include "cxplat.h"
+#include "tags.h"
+
 #include <memory.h>
 #include <string.h>
 
-#define CXPLAT_DEFAULT_TAG 'lpxc'
-
-__drv_allocatesMem(Mem) _Must_inspect_result_ _Ret_writes_maybenull_(size) void* cxplat_allocate(size_t size)
-{
-    return cxplat_allocate_with_tag(CxPlatNonPagedPoolNx, size, CXPLAT_DEFAULT_TAG, true);
-}
-
-__drv_allocatesMem(Mem) _Must_inspect_result_ _Ret_writes_maybenull_(new_size) void* cxplat_reallocate(
-    _In_ _Post_invalid_ void* memory, size_t old_size, size_t new_size)
-{
-    return cxplat_reallocate_with_tag(memory, old_size, new_size, CXPLAT_DEFAULT_TAG);
-}
-
-__drv_allocatesMem(Mem) _Must_inspect_result_
-    _Ret_writes_maybenull_(size) void* cxplat_allocate_cache_aligned(size_t size)
-{
-    return cxplat_allocate_cache_aligned_with_tag(size, CXPLAT_DEFAULT_TAG);
-}
-
 _Must_inspect_result_ _Ret_maybenull_z_ char*
 cxplat_duplicate_string(_In_z_ const char* source)
 {
     size_t size = strlen(source) + 1;
-    char* destination = (char*)cxplat_allocate(size);
+    char* destination = (char*)cxplat_allocate(CxPlatNonPagedPoolNx, size, CXPLAT_TAG_STRING, true);
     if (destination) {
         memcpy(destination, source, size);
     }
     return destination;
 }
 
+void
+cxplat_free_string(_Frees_ptr_opt_ _In_z_ const char* source)
+{
+    cxplat_free((void*)source, CXPLAT_TAG_STRING);
```

Comment:
I've always found it annoying that `free` functions don't take `const` pointers, requiring you to cast like this.